include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -3260,6 +3296,43 @@ namespace internal
 
 
 
+  template <int dim, int length, typename Number2, typename Number>
+  inline DEAL_II_ALWAYS_INLINE void
+  do_test_xy(AlignedVector<Number2> &                    values,
+             const std::vector<unsigned int> &           renumber,
+             const dealii::ndarray<Number, 200, 2, dim> &shapes,
+             const Number2 &                             test_grad_x,
+             const Number2 &                             test_grad_y,
+             const Number2 &                             test_value_z,
```

Comment:
To get this more compact, and make it more similar to the `do_interpolate_xy`, you could also collect these three values in an `std::array<Number2, 3>`. This would reduce the arguments to pass in, possibly making them fit in a single line.

Reply:
Done and yes it fits! I have added comments on which entry is which.